pkg/generate/crd/crd.go

```diff
@@ -86,70 +78,57 @@ func (c *Generator) ValidateAndInitFields() error {
 		return fmt.Errorf("error validating apis path %s: %v", apisPath, err)
 	}
 
-	c.apisPkg, err = crdutil.DirToGoPkg(apisPath)
+	c.apisPkg, err = util.DirToGoPkg(apisPath)
 	if err != nil {
 		return err
 	}
 
-	// Init output directory
 	if c.OutputDir == "" {
 		c.OutputDir = path.Join(c.RootPath, "config/crds")
 	}
 
 	return nil
 }
 
-// Do manages CRD generation.
-func (c *Generator) Do() error {
-	arguments := args.Default()
-	b, err := arguments.NewBuilder()
+// Run generates CRD and stores CRD into output dir.
+func (c *Options) Run() error {
```

Comment:
I found the previous abstractions easier to grok. 
Generator takes input options to generate CRD vs Options having Run method to generate the CRD. I find previous one more intuitive.

Reply:
I see. I will rebase and update the PR. How about we do it by a series of changes. First, I just focus on moving packages in this PR, and then refactor the `Generator` ?